src/core/tiledscene/qgstiledscenetexturecolorrenderer.cpp

```diff
@@ -0,0 +1,87 @@
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgstiledscenetexturecolorrenderer.h"
+
+QgsTiledSceneTextureColorRenderer::QgsTiledSceneTextureColorRenderer()
+{
+
+}
+
+QString QgsTiledSceneTextureColorRenderer::type() const
+{
+  return QStringLiteral( "texturecolor" );
+}
+
+QgsTiledSceneRenderer *QgsTiledSceneTextureColorRenderer::clone() const
+{
+  std::unique_ptr< QgsTiledSceneTextureColorRenderer > res = std::make_unique< QgsTiledSceneTextureColorRenderer >();
+
+  copyCommonProperties( res.get() );
+
+  return res.release();
+}
+
+QgsTiledSceneRenderer *QgsTiledSceneTextureColorRenderer::create( QDomElement &element, const QgsReadWriteContext &context )
+{
+  std::unique_ptr< QgsTiledSceneTextureColorRenderer > r = std::make_unique< QgsTiledSceneTextureColorRenderer >();
+
+  r->restoreCommonProperties( element, context );
+
+  return r.release();
+}
+
+QDomElement QgsTiledSceneTextureColorRenderer::save( QDomDocument &doc, const QgsReadWriteContext &context ) const
+{
+  QDomElement rendererElem = doc.createElement( QStringLiteral( "renderer" ) );
+
+  rendererElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "texturecolor" ) );
+
+  saveCommonProperties( rendererElem, context );
+
+  return rendererElem;
+}
+
+Qgis::TiledSceneRendererFlags QgsTiledSceneTextureColorRenderer::flags() const
+{
+  return Qgis::TiledSceneRendererFlag::RequiresTextures;
+}
+
+void QgsTiledSceneTextureColorRenderer::renderTriangle( QgsTiledSceneRenderContext &context, const QPolygonF &triangle )
+{
+  const QImage textureImage = context.textureImage();
+  if ( textureImage.isNull() )
+    return;
+
+  double textureX1;
+  double textureY1;
+  double textureX2;
+  double textureY2;
+  double textureX3;
+  double textureY3;
+  context.textureCoordinates( textureX1, textureY1, textureX2, textureY2, textureX3, textureY3 );
+
+  const QColor centerColor( textureImage.pixelColor(
+                              ( ( textureX1 + textureX2 + textureX3 ) / 3 ) * ( textureImage.width() - 1 ),
+                              ( ( textureY1 + textureY2 + textureY3 ) / 3 ) * ( textureImage.height() - 1 ) )
+                          );
+  QBrush b( centerColor );
+  QPainter *painter = context.renderContext().painter();
+  painter->setPen( Qt::NoPen );
+  painter->setBrush( b );
+  painter->drawPolygon( triangle );
+}
```

Comment:
Is the texture color renderer much faster than the "true" texture renderer? I like the artistic effect of this renderer, but I am wondering if it will be generally useful - any situations when someone would want to use this one instead of full texture renderer?

Reply:
From the PR description: "Gives a clear visual overview of the mesh structure while also providing some context via the texture color". It's kind of a trade off between the full texture view (which doesn't reveal the data structure) vs the complete loss of coloring information which the mesh/wireframe renderer gives.

I was toying with just making this a checkbox option in the texture renderer though. Or alternatively it could be a option for the wireframe renderer, so that users could color their individually crafted fill symbol with the representative color...

Reply:
I've moved this option to the wireframe renderer